test_app/lib/screens/test_buttons/file_presentation_service_test_buttons.dart

```diff
@@ -113,92 +118,114 @@ class _FilePresentationServiceTestButtonsState
     );
   }
 
-  void convert() async {
-    FilePickerResult? result = await FilePicker.platform.pickFiles(
-      type: FileType.custom,
-      allowedExtensions: ['doc', 'docx', 'ppt', 'pptx', 'pdf'],
-    );
-    String? path = result?.files.single.path;
-
-    if (path != null) {
-      _dolbyioCommsSdkFlutterPlugin.filePresentation
-          .convert(File(path))
-          .then((value) {
-        fileConverted = value;
-        return showDialog(context, 'Success', value.toJson().toString());
-      }).onError((error, stackTrace) =>
-              showDialog(context, 'Error', error.toString()));
-    } else {
-      Fluttertoast.showToast(msg: "File not selected");
+  Future<void> convert() async {
+    try {
+      FilePickerResult? result = await FilePicker.platform.pickFiles(
+        type: FileType.custom,
+        allowedExtensions: ['doc', 'docx', 'ppt', 'pptx', 'pdf'],
+      );
+      String? path = result?.files.single.path;
+
+      if (path != null) {
+        var fileConverted = await _dolbyioCommsSdkFlutterPlugin.filePresentation
+            .convert(File(path));
+        _fileConverted = fileConverted;
+        if (!mounted) return;
+        showDialog(context, 'Success', 'OK');
+      } else {
+        if (!mounted) return;
+        showDialog(context, 'File not selected', '');
+      }
```

Comment:
Would following work in stead:

```
if (path == null) {
  if (!mounted) return;
  showDialog(context, 'File not selected', '');
  return
}

var fileConverted = await _dolbyioCommsSdkFlutterPlugin.filePresentation
             .convert(File(path));
 _fileConverted = fileConverted;
if (!mounted) return;
showDialog(context, 'Success', 'OK');
```

What do you think?

Nevertheless in general it seems that we have a lot of repetition in the button handers. Would it be a way of eliminating it ?